core/src/main/java/org/elasticsearch/common/network/Cidrs.java

```diff
@@ -113,4 +113,9 @@ static String octetsToCIDR(int[] octets, int networkMask) {
         assert octets.length == 4;
         return octetsToString(octets) + "/" + networkMask;
     }
+
+    public static String createCIDR(long ipAddress, int networkMask) {
+        long blockSize = 1L << (32 - networkMask);
+        return octetsToCIDR(longToOctets(ipAddress - (ipAddress & (blockSize - 1))), networkMask);
+    }
```

Comment:
if we keep it here, maybe we should at least require an address that is valid for the given mask like cidrMaskToMinMax expects?